lib/Parse/Lexer.cpp

```diff
@@ -1884,23 +1884,12 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
         // Successfully scanned the body of the expression literal.
         ++CurPtr;
         continue;
+      } else if (!isMultilineString) {
+        diagnose(CurPtr, diag::lex_unterminated_string);
+        diagnose(CurPtr, diag::string_interpolation_unclosed);
       } else if (*CurPtr == '\r' || *CurPtr == '\n') {
-        if (IsMultilineString) {
-          // The only case we reach here is unterminated single line string in the
-          // interpolation. For better recovery, go on after emitting an error.
-          diagnose(CurPtr, diag::string_interpolation_unclosed);
-          diagnose(--TmpPtr, diag::opening_paren);
-          if (CurPtr != BufferEnd) diagnose(CurPtr, diag::lex_unterminated_string);
-          wasErroneous = true;
-          continue;
```

Comment:
We need this recovery epilogue for the for the `*CurPtr == '\r' || *CurPtr == '\n'` condition too (and the above comment).